src/Simplify.elm

```diff
@@ -6556,25 +6556,34 @@ listConcatChecks =
             (\checkInfo ->
                 case fromListGetLiteral listCollection checkInfo.lookupTable checkInfo.firstArg of
                     Just listLiteral ->
-                        if List.all AstHelpers.isListLiteral listLiteral.elements then
+                        let
+                            (Node firstArgRange _) =
+                                checkInfo.firstArg
+                        in
+                        if
+                            (firstArgRange.start.row /= firstArgRange.end.row)
```

Comment:
I'd prefer
```elm
not (List.all (\(Node itemRange _) -> itemRange.end.row == firstArgRange.start.row) listLiteral.elements)
```
instead of
```elm
(firstArgRange.start.row /= firstArgRange.end.row)
```
to account for e.g.
```elm
List.concat
    [ [ 1, 2 ]
    ]
```
A test for that case would be nice as well